extensions/git/src/git.ts

```diff
@@ -610,6 +611,14 @@ export function parseGitCommit(raw: string): Commit | null {
 	return { hash: match[1], message: match[5], parents, authorEmail: match[2] };
 }
 
+export function parseGitFilterAttribute(raw: string): string | null {
+	const parts = raw.trim().split('\0');
+	if (parts.length < 3 || parts[1] !== 'filter' || parts[2] === 'unspecified') {
+		return null;
+	}
+	return parts[2];
+}
```

Comment:
Note for other reviewers:
I thought this was wrong at first by expecting the filter to be the first attribute to be `filter`, but if you see line 821, you'll see that this is the output of `git check-attr` specifying the `filter` attribute (rather than getting the raw line in `.giattributes` [which was discussed in the PR description]), so this works generically.

Reply:
The PR description could probably be worded better but the intention was to point out we will use `git check-attr` to parse `.gitattributes` (which you need to create/modify to support `git-crypt`) instead of doing this ourselves. This saves worrying about edge cases, format of file changing etc. Git gives us a good interface for this so lets use it.

Reply:
I also didn't thoroughly read the PR >.>